### PR TITLE
Add Tailscale Serve/Funnel support for remote access

### PR DESCRIPTION
**Pull request by fwdslsh-dev**
## Summary

Adds Tailscale Serve/Funnel support so the assistant and/or guardian can be reached from outside the home network, without opening a router port. This is the config model, compose plumbing, and control-plane apply logic — a `remote` addon that is off by default and private (Serve, not Funnel) when first enabled.

Scope note: the API route, UI card, and live tunnel status reads are deliberately **not** in this PR. Nothing yet calls `reconcileRemoteAccess` from a user action; it runs on install/update and on an addon credentials save.

### Config model (`remote-access.ts`)
- `RemoteAccessConfig`: `hostname` (pinned tailnet node name), `public` (Serve vs Funnel), `target` (assistant/guardian/both).
- Hostname derived from the compose project name with DNS-label validation (lowercase, hyphen-normalized, 63-char limit). Docker already forces project names to be unique per host, which makes multi-stack tailnet names collision-free by construction.
- `resolveServeConfig()` derives Tailscale's `ipn.ServeConfig` document. `AllowFunnel` keys are always emitted with an explicit boolean — a missing key reads to Tailscale as "no change", so omitting it would leave a previously-funneled service public indefinitely.
- Browser-safe (no `node:*`), so the setup wizard can import it directly.

### Apply logic (`remote-apply.ts`)
- `writeServeConfig()` writes the generated document atomically; `pinRemoteHostname()` persists the derived hostname write-once, so a later project rename cannot move the operator's public URL out from under existing bookmarks.
- `reconcileRemoteAccess()` is the single entry point: enabled pins and writes the live document, disabled writes the explicit empty one. It never deletes the file.

### Access toggles
- `remoteRequiresGuardianIngress()` lets the tunnel require the guardian's direct listener without opening the LAN bind — it reaches the guardian over `portal_net`, not through `OP_GUARDIAN_BIND_ADDRESS`. `guardianNetwork` stays the only source for the LAN bind itself.
- `resolveAccessEnv()`'s new option is optional, and omitting it reproduces pre-`remote` behavior byte-for-byte (pinned by a test over all 16 toggle combinations).

### Compose (`services.compose.yml`)
- `tunnel` service, profile-gated on `addon.remote`, digest-pinned, rootless, userspace networking (no `NET_ADMIN`, no `/dev/net/tun`, works on Docker Desktop).
- Publishes no host ports, so enabling remote access can never widen LAN exposure. No `depends_on`, because naming the profile-gated guardian would be a project-wide Compose parse error.
- On `assistant_net` and `portal_net` as an ingress path for the two first-party services it fronts — the same per-service exception `ollama` has.

### Secrets
- `TS_AUTHKEY` is delivered as a Compose file secret (`file:` prefix, so it never lands in `docker inspect`) and classified as delegated: a tailnet join key can enroll an arbitrary device onto the operator's tailnet, so it stays out of the assistant-visible stash.

### Tests
`remote-access.test.ts`, `remote-apply.test.ts`, `remote-compose.test.ts`, `remote-addon-registry.test.ts`, plus extensions to `access-toggles.test.ts`, `access-apply.test.ts`, `setup.test.ts`, `home.layout.test.ts`, and `install-edge-cases.test.ts`.

## Review fixes (863aabf)

A review of this branch found six issues, all fixed in the latest commit:

- **Removed `ipnserve.go`** — 1874 lines of verbatim upstream Tailscale source that landed accidentally in a docs commit. It was never part of the implementation.
- **`ts_authkey` is now seeded by `ensureSecrets`.** Compose fails container creation when a declared secret's file is missing, so enabling `remote` without first visiting the credentials form broke `compose up` for the entire stack. Seeded empty — blank is what selects interactive login.
- **Moved the generated serve config to `state/remote`.** It was under `system/stack/`, which is replaced wholesale from the release skeleton on update; the config and the directory containerboot watches were being destroyed by the next update, and containerboot hard-fails when it cannot register that watch.
- **`reconcileRemoteAccess` now runs in `applyHome`**, before any compose command, so the serve document exists and the hostname is pinned before the tunnel is ever created.
- **`performSetup` threads `guardianIngressRequired`.** A setup rerun previously recomputed `GUARDIAN_DIRECT_INGRESS` from the toggles alone, silently switching off the listener a live guardian-targeting tunnel depends on.
- **The credentials route regenerates the serve document before recreating the tunnel**, so saving remote config is no longer a setting that silently does nothing.

## Still to come

Live tunnel status reads, the API route, and the UI card. None of this has been exercised against a real Docker host yet — rootless + `TS_SOCKET` and SSE-through-Funnel are reasoned from upstream source, not observed.

https://claude.ai/code/session_01QheXMSCMYMLquHrtPkBYc7

### CODE REVIEW

File: packages/skeleton/system/stack/services.compose.yml [hunk 1]
```diff
@@ -3,13 +3,17 @@
...
-# (127.0.0.1:8880), so voice stays on addon_net only here. An operator who turns
-# on OP_VOICE_LAN_ACCESS opts into the same per-service exception ollama has —
+# OpenCode API (:4096) over a shared bridge. The exceptions are ollama and
+# tunnel. ollama is reached by the assistant as its LLM provider over the
+# Docker DNS name ollama:11434 — that inference path needs shared-network
+# reachability, so ollama is granted assistant_net per service. tunnel is not
+# a general addon at all: it is an INGRESS PATH for the two first-party
+# services it fronts (assistant, guardian), never a third party's code, so it
+# gets the same per-service exception for the same reason — see the tunnel
+# service below. voice needs no assistant reachability at all BY DEFAULT: the
```
[the remark this answers is not in the document]
Comment (fwdslsh-dev) [reply packages/skeleton/system/stack/services.compose.yml]: Valid — fixed in f2ff967. "Never a third party's code" was plainly false (`tailscale/tailscale` is upstream's image), and it was doing rhetorical work in the justification for a trust-boundary exception, which is the worst place for an overstatement.

Reworded to say what's actually true: tunnel is first-party-*chosen*, not first-party-*written*, and what bounds the exception is blast radius rather than authorship — digest-pinned image, rootless, zero added capabilities, zero device mounts, no published host port, one delegated secret. Also noted that the same caveat applies to `ollama`'s pre-existing exception directly above, since it has the same authorship gap and the old wording implied a distinction that wasn't there.

The matching comment in `network-partitioning.test.ts` drew the same not-a-third-party distinction; it now says "operator-installed addon", which is the distinction that actually holds.

---
_Generated by [Claude Code](https://claude.ai/code)_

**Comment by itlackey** [comment]
@codex review this PR for design corretness and over engineering.

File: packages/lib/src/control-plane/remote-apply.ts [hunk 2]
```diff
@@ -0,0 +1,207 @@
...
+ * Follows `access-apply.ts`'s convention: never throws. A failure in either
+ * step is caught and surfaced via `error`, with the rest of the result
+ * falling back to safe defaults rather than partially-read state.
+ */
+export function reconcileRemoteAccess(homeDir: string): RemoteAccessReconcileResult {
+  try {
+    const { enabled, config } = readRemoteAccessState(homeDir);
+
+    if (!enabled) {
+      writeServeConfigDoc(homeDir, DISABLED_SERVE_CONFIG);
```
[the remark this answers is not in the document]
Comment (fwdslsh-dev) [reply packages/lib/src/control-plane/remote-apply.ts:217]: Confirmed and fixed in 14ef96b. `setAddonEnabled` (addons.ts) never called the reconcile — verified by reading it, not just by inspection of the call graph — so both halves of this were live: enable started `tunnel` against the previous document, and disable left the live document on disk while only attempting a container stop.

The reconcile now runs **inside** the shared toggle, and specifically *before* the caller starts or stops anything. That ordering is the fail-closed property you're describing: on disable the empty document is on disk by the time any `compose stop` runs, so a stop that fails can't leave a Funnel reachable. There's a test pinning exactly that (`writes the disabled serve document when remote is disabled`).

One implementation note, since it's visible in the diff: `addons.ts` now imports `remote-apply.ts`, and `remote-apply.ts` was importing `listEnabledAddonIds` from `addons.ts` — a cycle. `addon-ids.ts` is documented as a pure-constants file precisely so both sides can import it, so the enabled-addon read is rebuilt in `remote-apply.ts` from the same three leaf primitives `listEnabledAddonIds` uses (`readStackEnv`, `parseEnabledAddons`, `BUILTIN_ADDON_IDS`). The parse stays shared; only the wrapper is duplicated, and a table-driven test asserts the two agree on the cases that could diverge (padding, duplicates, unknown ids, empty) so they can't drift silently.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: packages/ui/src/routes/api/host/addons/[name]/credentials/+server.ts [hunk 3]
```diff
@@ -223,6 +224,27 @@ export const POST: RequestHandler = async (event) => {
...
 
+    // The `remote` addon is the one built-in whose config is not read from
+    // stack.env by its own container at all: `tunnel` reads a GENERATED
+    // Tailscale serve document, so persisting OP_REMOTE_TARGET/OP_REMOTE_PUBLIC
+    // without regenerating that document would recreate the container below
+    // only for it to re-read the previous config — a saved setting that
+    // silently does nothing. Regenerate first, then let the recreate pick it
+    // up. reconcileRemoteAccess never throws; it reports failure in its result.
+    if (name === 'remote') {
+      const remote = reconcileRemoteAccess(state.homeDir);
```
[the remark this answers is not in the document]
Comment (fwdslsh-dev) [reply packages/ui/src/routes/api/host/addons/[name]/credentials/+server.ts]: Confirmed and fixed in 14ef96b. `remoteRequiresGuardianIngress` had exactly two call sites — `access-apply.ts` and `setup.ts` — and neither is on the credentials path, so a target change through the addon drawer set the ServeConfig and nothing else. You're also right that the recreate scope couldn't have covered it: `ADDON_ENV_RECREATE_SCOPE` is keyed by env var and maps `OP_REMOTE_*` to `tunnel`, and "…and guardian, but only if this save flipped ingress" depends on prior state, not on which key changed — it isn't expressible in that table.

New `applyRemoteAccess()` does the reconcile, the `GUARDIAN_DIRECT_INGRESS` recomputation, and the resulting recreate set as one unit; the credentials route and `setAddonEnabled` both call it, and the route unions its returned services into the scope.

Two deliberate limits, both tested:

- It writes `GUARDIAN_DIRECT_INGRESS` and **nothing else** in the access env. `resolveAccessEnv` is the same helper `applyAccessToggles` uses, so `guardianNetwork` remains the sole source of the LAN bind — enabling remote access still cannot widen LAN exposure, and disabling remote won't switch ingress off when `guardianNetwork` independently wants it on.
- On the third case you raise — no guardian deployed at all — it returns a **warning** rather than auto-enabling an ingress addon. Enabling a portal deploys a new network-listening service, which reads to me as the operator's call, not a side effect of saving a target. The warning names the addons that would deploy one and is surfaced by both callers. Happy to make it auto-enable instead if you'd rather it match `reconcileGuardianIngressAddons`' behavior on the access-toggles path.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: packages/lib/src/control-plane/addon-env-schemas.ts [hunk 4]
```diff
@@ -117,6 +134,41 @@ SLACK_BOT_NAME=
...
+  remote: `# OpenPalm Remote Access (Tailscale tunnel) configuration
+# ---
+# Lets you reach this assistant when you're away from home, without opening
+# any ports on your router. Values are optional; the tunnel container supplies
+# safe defaults.
+
+# What the tunnel points at: assistant, guardian, or both. Most people only
+# ever need "assistant" — the guardian target is for advanced setups that also
+# expose bot/API portals remotely.
+OP_REMOTE_TARGET=assistant
```
[the remark this answers is not in the document]
Comment (fwdslsh-dev) [reply packages/lib/src/control-plane/addon-env-schemas.ts:146]: Confirmed and fixed in 14ef96b — and it's worse than described in two ways worth recording.

**It's pre-existing and general, not remote-specific.** `value: ""` predates this PR (`git log -L` puts the current form at 62dbf91, which only added the boolean carve-out). Every addon with a non-sensitive text field has been silently clearing untouched fields on save — `SLACK_BOT_NAME`, the voice fields, all of them. This PR doesn't introduce the bug; it's the first place the bug redirects a network tunnel.

**`OP_REMOTE_HOSTNAME` is the sharper case.** Blanking the target falls back to `assistant`, as you say. Blanking the hostname un-pins it — `pinRemoteHostname` treats empty as unpinned and re-derives from the project name — so an operator who set a custom tailnet name loses it and their public URL moves. That's precisely the failure the write-once pin exists to prevent, reintroduced through the form.

Fixed on the GET side rather than per-addon, since the round-trip is what's broken: a field that **is** set now returns its value; `@sensitive` still returns `""` (a stored `TS_AUTHKEY` must never reach the DOM); **unset** fields still return `""` so a save doesn't materialize every schema default into `stack.env` as an explicit setting — booleans keep their default-when-unset, since an unchecked box has to render some concrete state.

Three tests: a set field round-trips, a sensitive field doesn't, and a replay of what the drawer actually does (seed from GET → change one field → submit all) asserting `OP_REMOTE_TARGET` and `OP_REMOTE_HOSTNAME` survive a save that only touched `OP_REMOTE_PUBLIC`.

---
_Generated by [Claude Code](https://claude.ai/code)_
